ledger/simulation/tracer.go

```diff
@@ -245,6 +252,26 @@ func (tracer *evalTracer) makeOpcodeTraceUnit(cx *logic.EvalContext) OpcodeTrace
 	return OpcodeTraceUnit{PC: uint64(cx.PC())}
 }
 
+func (o *OpcodeTraceUnit) appendDeletedStackValue(cx *logic.EvalContext, tracer *evalTracer) {
+	tracer.stackChangeExplanation = cx.NextStackChange()
+
+	stackHeight := len(cx.Stack)
+	stackHeightAfterDeletion := stackHeight - tracer.stackChangeExplanation.Deletions
+	tracer.stackHeightAfterDeletion = stackHeightAfterDeletion
+
+	if tracer.stackChangeExplanation.Deletions == 0 {
+		return
+	}
+	for i := stackHeightAfterDeletion; i < stackHeight; i++ {
+		tealValue := cx.Stack[i].ToTealValue()
+		o.Deleted = append(o.Deleted, model.TealValue{
+			Type:  uint64(tealValue.Type),
+			Uint:  tealValue.Uint,
+			Bytes: tealValue.Bytes,
```

Comment:
Since `Bytes` is a string, this assignment will cause us to report raw binary data as a JSON string, which we need to avoid. All binary data should be base64 encoded when JSON encoding (and preferably NOT base64 encoded when msgpack encoding).

Unfortunately it looks like `TealValue` was declared in our REST API as just a string, not binary data, so it does not have these properties...

Reply:
good find!  I was unaware of the b64 encoding for TEAL value, but that makes sense.

I updated the testcases and encoding in [convert to b64 on stack change var](https://github.com/algorand/go-algorand/pull/5440/commits/7b42650146074e97357adabc29599532496f7156)

Reply:
I was thinking more on this, and felt a bit off to convert at ledger level. It feels better to convert at REST level.

See [move b64 conversion to rest endpoint export](https://github.com/algorand/go-algorand/pull/5440/commits/05ecc8b6fe55175f1dc883bdead927b7c8da788a)